runners/requirements.go

```diff
@@ -0,0 +1,69 @@
+package runners
+
+import (
+	"encoding/json"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"golang.org/x/sys/cpu"
+
+	"github.com/ollama/ollama/gpu"
+)
+
+type RunnerRequirements struct {
+	SystemInfo  string   `json:"system_info"`
+	Version     string   `json:"version"`
+	CPUFeatures []string `json:"cpu_features"`
+}
+
+func GatherRequirements(exe string) (RunnerRequirements, error) {
```

Comment:
```suggestion
func gatherRequirements(exe string) (RunnerRequirements, error) {
```

if not used outside of this package